### DISCUSSION TITLE
Alternative to react-router location.state and location.hash in next js

### DISCUSSION THREAD

**Discussion by hammad-shahid-educative** [vercel/next.js#13924] [Help]
location in react- router looks like this:
```
{
  key: 'ac3df4', // not with HashHistory!
  pathname: '/somewhere',
  search: '?some=search-string',
  hash: '#howdy',
  state: {
    [userDefined]: true
  }
}
```
previously with react-router I can pass any value in state using history.push and that values are not visible in url. How can I do it with next js ? Also how can I get hash values in next js

**Comment by Rakeshtagadghar** [comment]
@timneutkens  It will be a good feature if we can add it in nextjs as passing data from react-router is built-in persistent and not visible in url. This will be helpful for server side rendering async data , when we dont want to expose sensitive information by storing in cookie/localstorage/sessionstorage
